packages/twenty-server/src/engine/twenty-orm/repository/workspace.repository.ts

```diff
@@ -565,20 +571,39 @@ export class WorkspaceRepository<
         ]);
       }),
     );
+    const relationMetadataArgsMap = new Map(
+      relationMetadataArgsCollection.map((relationMetadataArgs) => [
+        relationMetadataArgs.name,
+        relationMetadataArgs,
+      ]),
+    );
     const newData: object = {};
 
     for (const [key, value] of Object.entries(data)) {
       const compositePropertyArgs = compositeFieldMetadataArgsMap.get(key);
+      const relationMetadataArgs = relationMetadataArgsMap.get(key);
 
-      if (!compositePropertyArgs) {
-        if (typeof value === 'object') {
+      if (!compositePropertyArgs && !relationMetadataArgs) {
+        if (isPlainObject(value)) {
           newData[key] = this.formatResult(value);
         } else {
           newData[key] = value;
         }
         continue;
       }
 
+      if (relationMetadataArgs) {
```

Comment:
We can't stick to just one level of formatting result, as we'll often need to use relations.
So we're now recursively formatting them based on the `relationMetadataArgs`.
We should be careful with that, and properly test performance behaviour.
In my test I didn't get real difference before and after.